### PR TITLE
coq: update to 8.12.1

### PR DESCRIPTION
**Pull request by smorimoto**

### CODE REVIEW

**Comment by ericonr** [comment]
Please rebase. `coccinelle` was rebuilt recently as well.

**Comment by smorimoto** [comment]
The PR title doesn't match the changes because commits were pushed to the master directly. ​I can make the title clearer, but I will leave it as it is for now.

File: srcpkgs/coccinelle/template [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'coccinelle'
 pkgname=coccinelle
 version=1.0.8
-revision=5
+revision=6
 build_style=gnu-configure
```
Comment (ericonr) [inline srcpkgs/coccinelle/template:5]: Is the revbump necessary, now that `ocaml-num` no longer changes?

Comment (smorimoto) [reply srcpkgs/coccinelle/template:5]: Oh indeed.

**Pull request merged**